### PR TITLE
add purge-config loader

### PR DESCRIPTION
**Pull request by gibkigonzo**
### Short Description and Why It's Useful

Adds loader that clears properties from config in build. Those properties list can be extended by adding to config.purgeConfig (**add properties that are used only for `server.ts` and related**) 

### Which Environment This Relates To
Check your case. In case of any doubts please read about [Release Cycle](https://docs.vuestorefront.io/guide/basics/release-cycle.html)

- [ ] Test version (https://test.storefrontcloud.io) - this is a new feature or improvement for Vue Storefront. I've created branch from `develop` branch and want to merge it back to `develop`
- [ ] RC version (https://next.storefrontcloud.io) - this is a stabilisation fix for Release Candidate of Vue Storefront. I've created branch from `release` branch and want to merge it back to `release`
- [x] Stable version (https://demo.storefrontcloud.io) - this is an important fix for current stable version. I've created branch from `hotfix` or `master` branch and want to merge it back to `hotfix`

### Upgrade Notes and Changelog
- [x] No upgrade steps required (100% backward compatibility and no breaking changes)
- [x] I've updated the [Upgrade notes](https://github.com/DivanteLtd/vue-storefront/blob/develop/docs/guide/upgrade-notes/README.md) and [Changelog](https://github.com/DivanteLtd/vue-storefront/blob/develop/CHANGELOG.md) on how to port existing Vue Storefront sites with this new feature

**IMPORTANT NOTICE** - Remember to update `CHANGELOG.md` with description of your change

### Contribution and Currently Important Rules Acceptance

- [x] I read and followed [contribution rules](https://github.com/DivanteLtd/vue-storefront/blob/master/CONTRIBUTING.md)

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -5,6 +5,12 @@ All notable changes to this project will be documented in this file.
 The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/)
 and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).
```
Comment (pkarw) [inline CHANGELOG.md:7]: 1.12.1 @gibkigonzo

Comment (gibkigonzo) [reply CHANGELOG.md:7]: done :ok_hand:

**Comment by Fifciu** [comment]
@gibkigonzo could you please add `cloudflare` to `purgeConfig` array in config? I believe it would prevent more people from publishing Cloudflare key unconsciously.

**Comment by gibkigonzo** [comment]
@Fifciu sure, I will add this :ok_hand:

**Pull request merged**